### PR TITLE
Test aggregates

### PR DESCRIPTION
**Pull request by rivudhk**
Is this a user-visible change (yes/no): no

### CODE REVIEW

**Comment by rivudhk** [comment]
Partially fixes #1390

File: .vscode/settings.json [hunk 1]
```diff
@@ -26,5 +26,14 @@
     "prettier.requireConfig": true,
     "rust-analyzer.showUnlinkedFileNotification": false,
     "svelte.ask-to-enable-ts-plugin": false,
-    "svelte.enable-ts-plugin": true
+    "svelte.enable-ts-plugin": true,
+    "python.testing.unittestArgs": [
```
Comment (mihaibudiu) [inline .vscode/settings.json:30]: is this supposed to be checked in?

File: python/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (mihaibudiu) [inline python/.vscode/settings.json:1]: same question

File: python/tests/test_aggregates.py [hunk 3]
```diff
@@ -0,0 +1,335 @@
+import unittest
+from feldera import SQLContext
+from tests import TEST_CLIENT
+
+class TestAggregates(unittest.TestCase):
```
Comment (mihaibudiu) [inline python/tests/test_aggregates.py:5]: let's structure this use classes.
There should be a base class TestAggregatesBase, and all tests derive from this class.
The class has fields for the table, the view, the input data, and the expected results
I think the table is the same in all tests, so it should be part of the base class too
The code to run the test is entirely in the base class:
```
        out = sql.listen("avg_view_value")
        sql.start()
        sql.input_json("avg0_value", data)
        sql.wait_for_completion(True)
        out_data = out.to_dict()
        for datum in expected_data:
            datum.update({"insert_delete": 1})
        assert expected_data == out_data
```

Comment (mihaibudiu) [reply python/tests/test_aggregates.py:5]: So the code here will become one base class and 4 derived classes, for AVG, AVG DISTINCT, AVG WHERE, and AVG GROUPBY

File: python/tests/test_aggregates.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+                                       FROM avg0_distinct''')
+        data = [
+            {"c1": 1, "c2": 2, "c3": 3, "c4": 4, "c5": 5, "c6": 6, "c7": None, "c8": 8},
+            {"c1": None, "c2": 2, "c3": 30, "c4": 4, "c5": None, "c6": 60, "c7": 70, "c8": 8},
+            {"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        
+        out = sql.listen("avg_view_distinct")
+        
+        sql.start()
```
Comment (mihaibudiu) [inline python/tests/test_aggregates.py:158]: you like empty lines a lot. I would use empty lines only infrequently. space on the screen is an important resource

File: python/tests/test_aggregates.py [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+                                            AVG(c1) AS c1,
+                                            AVG(c2) AS c2,
+                                            AVG(c3) AS c3,
+                                            AVG(c4) AS c4,
+                                            AVG(c5) AS c5,
+                                            AVG(c6) AS c6,
+                                            AVG(c7) AS c7,
+                                            AVG(c8) AS c8
+                                       FROM avg0_where
+                                       WHERE
```
Comment (mihaibudiu) [inline python/tests/test_aggregates.py:252]: here I would also add a different predicate - so two instances of this test.

Comment (rivudhk) [reply python/tests/test_aggregates.py:252]: Does this mean I should create a different test or add one more condition?

Comment (mihaibudiu) [reply python/tests/test_aggregates.py:252]: Let's do the other refactoring first

**Comment by mihaibudiu** [comment]
I don't think we should merge this yet

**Comment by mihaibudiu** [comment]
Please open a new PR where we can review this file. You can leave this commit merged, probably we'll overwrite it.

**Comment by rivudhk** [comment]
I mistakenly merged this. I apologize for the oversight. I'll create a new PR.

**Pull request merged**